buildkitd/buildkitd.go

```diff
@@ -356,3 +362,44 @@ func GetAvailableImageID(ctx context.Context, image string) (string, error) {
 	}
 	return string(output), nil
 }
+
+// CheckCompatibility runs all avaliable compatibility checks before starting the buildkitd daemon.
+func CheckCompatibility(ctx context.Context, settings Settings) error {
+	isNamespaced, err := isNamespacedDocker(ctx)
+	if isNamespaced {
+		return errors.New(`user namespaces are enabled, run with --bkd-arg "--userns=host" to disable`)
```

Comment:
Won't this fail even if the setting is enabled?

Reply:
We dont even check compatibility if you specify additional buildkit args; we assume you know what you are doing and are non-standard. It didn't seem worth setting a precedent for either handling each arg as a first-class citizen, or parsing the passed-in args.

Reply:
Ah yes. Good thinking

Reply:
Excuse me; we check, but it won't cause a failure. Lines 189-190.